Split/Api/DefaultSplitFactoryBuilder.swift

```diff
@@ -129,30 +131,30 @@ import Foundation
             validationLogger.log(errorInfo: errorInfo, tag: validationTag)
         }
 
-        let finalKey = Key(matchingKey: matchingKey ?? "", bucketingKey: bucketingKey)
+        params.key = Key(matchingKey: matchingKey, bucketingKey: bucketingKey)
 
         var factory: SplitFactory?
-        if apiKey.uppercased() == kApiKeyLocalhost {
-            factory = LocalhostSplitFactory(key: finalKey,
-                                            config: config ?? SplitClientConfig(),
+        if params.apiKey.uppercased() == kApiKeyLocalhost {
+            factory = LocalhostSplitFactory(key: params.key,
+                                            config: params.config,
                                             bundle: bundle)
         } else {
             do {
-                factory = try DefaultSplitFactory(apiKey: apiKey,
-                                                  key: finalKey,
-                                                  config: config ?? SplitClientConfig(),
-                                                  httpClient: httpClient,
-                                                  reachabilityChecker: reachabilityChecker,
-                                                  testDatabase: testDatabase,
-                                                  notificationHelper: notificationHelper)
+                factory = try DefaultSplitFactory(params)
             } catch ComponentError.notFound(let name) {
                 Logger.e("Component was not created properly: \(name)")
             } catch {
                 Logger.e("Error: \(error)")
             }
         }
 
-        DefaultSplitFactoryBuilder.factoryMonitor.register(instance: factory, for: apiKey)
+        DefaultSplitFactoryBuilder.factoryMonitor.register(instance: factory, for: params.apiKey)
+        factoryCount = DefaultSplitFactoryBuilder.factoryMonitor.instanceCount(for: params.apiKey)
+        let activeCount = DefaultSplitFactoryBuilder.factoryMonitor.activeCount()
+        if let telemetryStorage = params.telemetryStorage {
+            telemetryStorage.recordActiveFactories(count: activeCount)
+            telemetryStorage.recordRedundantFactories(count: factoryCount - 1)
```

Comment:
Is it worth double checking to prevent count from being `-1`?

Reply:
That's a good point. It shouldn't be less than 0, and if that is the case It will be good have that recorded to realize that some component is not working properly. Remember that this is information we track to check SDK functioning